src/main/java/com/iopipe/plugin/profiler/BufferPoolStatistics.java

```diff
@@ -0,0 +1,98 @@
+package com.iopipe.plugin.profiler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+
+/**
+ * This contains statistics on the buffer pool.
+ *
+ * @since 2108/05/27
+ */
+public final class BufferPoolStatistics
+{
+	/** The name of the pool. */
+	public final String name;
+	
+	/** The number of objects in the pool. */
+	public final long count;
+	
+	/** The memory used in the pool. */
+	public final long usedbytes;
+	
+	/** The capacity of the pool in bytes. */
+	public final long capacitybytes;
+	
+	/**
+	 * Initializes an unknown buffer pool statistic.
+	 *
+	 * @since 2018/05/30
+	 */
+	public BufferPoolStatistics()
+	{
+		this(null, -1, -1, -1);
+	}
+	
+	/**
+	 * Initializes the buffer pool statistics.
+	 *
+	 * @param __n The name of the pool.
+	 * @param __c The objects in the pool.
+	 * @param __u The number of used bytes.
+	 * @param __p The capacity of the pool.
+	 * @since 2018/05/27
+	 */
+	public BufferPoolStatistics(String __n, long __c, long __u, long __p)
```

Comment:
instead of `__x` names, could these be meaningful/written out as their full value (name, objects, usedBytes ...) to make this more readable?

Reply:
I can do that!